homework/App.js

```diff
@@ -12,18 +14,41 @@ class App {
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('label', header, { for: 'select-menu', text: 'HYF Repositories' });
+    const selectMenu = Util.createAndAppend('select', header, {
+      class: 'select-menu',
+      id: 'select-menu',
+    });
+    Util.createAndAppend('main', root, { id: 'container' });
 
     try {
-      const repos = await Util.fetchJSON(url);
+      const reposResponse = await fetch(url);
+      const repos = await reposResponse.json();
+      if (reposResponse.status !== 200) {
+        throw new Error(`Oops, error: ${reposResponse.status}, ${reposResponse.statusText}`);
+      }
       this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      this.repos
+        .sort((a, b) =>
+          a
+            .name()
+            .toLowerCase()
```

Comment:
I can't remember if I commented on this in earlier homework, but when using `localeCompare` you should not convert to lower case first. Some languages do not even have different cases (e.g. Arabic). All language-specific aspects for comparing two strings are completely handled by `localeCompare`, which uses the language set up by the user in the computer's operating system for its rules.